network/client_test.go

```diff
@@ -24,6 +24,9 @@ func testClient(t *testing.T, fac routerFactory, cl clientFactory) {
 		t.Fatal(err)
 	}
 	go r.Start()
+	for !r.Listening() {
+		time.Sleep(20 * time.Millisecond)
+	}
```

Comment:
Could this lead to an endless loop? Does it timeout if (for whatever reasons) we could never start listening?